.private/ci-build.sh

```diff
@@ -70,10 +77,48 @@ fi
 
 echo ""
 echo "Configuring ..."
+if [ "${tidy}" = "yes" ]; then
+# Clang-Tidy needs to be run with Clang compiler
+CFLAGS="${cflags}" CXXFLAGS="${cflags}" CC="clang" CXX="clang++" ../configure --enable-examples-build --enable-tests-build "$@"
+else
 CFLAGS="${cflags}" CXXFLAGS="${cflags}" ../configure --enable-examples-build --enable-tests-build "$@"
+fi
 
 echo ""
 echo "Building ..."
+
+if [ "${tidy}" = "yes" ]; then
+	# $(@D): '$(@D)' is equivalent to '$(dirname $@)'.
+	# $(<F): '$(<F)' is equivalent to '$(notdir $<)'.
+	# example: 'src/foo.c' -> 'builddir/src/foo.c.compdb.json'
+	# add CFLAGS here as configure escapes them
+	cflags+=" -MJ \$(@D)/\$(<F).compdb.json"
+	make -j4 -k CFLAGS="${cflags}" CXXFLAGS="${cflags}"
+
+	# Create compile_commands.json from all the .compdb.json files.
+	echo "[" > compile_commands.json
+	find . -name "*compdb.json" -exec cat {} \; >> compile_commands.json
+	echo "]" >> compile_commands.json
+
+	built_files=$(find . -name "*.compdb.json" -printf "${gitdir}/%P " | \
+		sed 's/.compdb.json//g')
+	reldr=$(realpath --relative-to="$(pwd)" "${gitdir}")
+
+
+	# Get latest clang-tidy-diff.py script from LLVM project.
```

Comment:
Well it is not the "latest" but a specific version, which I think is a wise choice.

And an empty line too much above.